clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/record/ValueRecord.java

```diff
@@ -113,4 +113,11 @@ public byte[] serialize() {
 
     return serializedArr;
   }
+
+  public byte[] getDataBytesPrependedWithWriterSchemaId() {
```

Comment:
Do we need this function? and it is doing bytes copying.
When combining multiple chunks, I think we can leave some headroom for the schema id, right?

Reply:
Hmm I am not sure if that is avoidable? 
Today I am using SingleGetChunkingAdapter and ValueRecord as the backbone for fetching the chunks and based on my understanding it will be added each chunk by calling SingleGetChunkingAdapter.addChunkIntoContainer() and after that it should call SingleGetChunkingAdapter.constructValue() which simply creates a ValueRecord container. I believe to create a assembled value, there will be at least one byte array allocated. I think before this particular method is called, no byte array is allocated, they are just wrapping the original byte array chunk...

Reply:
There are two types of RMD values, which can be retrieved from the database.
1. Non-Chunked RMD, so the returned result is `bytes[]`.
2. Chunked RMD, and the returned result should be a composited Bytebuf.

I think we can reuse the existing `serialize()` function, which will be able to make a smart decision about when to construct a new byte array, right?